hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -99,21 +99,33 @@ public NodeEngineImpl(final Node node) {
         ProbeLevel probeLevel = node.getGroupProperties().getEnum(PERFORMANCE_METRICS_LEVEL, ProbeLevel.class);
         this.metricsRegistry = new MetricsRegistryImpl(node.getLogger(MetricsRegistryImpl.class), probeLevel);
         this.proxyService = new ProxyServiceImpl(this);
-        this.serviceManager = new ServiceManagerImpl(this);
+        this.serviceManager = createServiceManager();
         this.executionService = new ExecutionServiceImpl(this);
         this.operationService = new OperationServiceImpl(this);
         this.eventService = new EventServiceImpl(this);
         this.waitNotifyService = new WaitNotifyServiceImpl(this);
         this.transactionManagerService = new TransactionManagerServiceImpl(this);
         this.wanReplicationService = node.getNodeExtension().createService(WanReplicationService.class);
-        this.packetDispatcher = new PacketDispatcherImpl(
+        this.packetDispatcher = createPacketDispatcher();
+        quorumService = new QuorumServiceImpl(this);
+    }
+
+    protected PacketDispatcher createPacketDispatcher() {
+        return new PacketDispatcherImpl(
                 logger,
                 operationService,
                 eventService,
                 wanReplicationService,
-                new ConnectionManagerPacketHandler()
+                connectionManagerPacketHandler()
         );
-        quorumService = new QuorumServiceImpl(this);
+    }
+
+    protected PacketHandler connectionManagerPacketHandler() {
+        return new ConnectionManagerPacketHandler();
+    }
+
+    protected ServiceManagerImpl createServiceManager() {
+        return new ServiceManagerImpl(this);
```

Comment:
I think we already solved problem with service registration, no need to extend service managers.